rust/lance-index/src/vector/pq.rs

```diff
@@ -478,7 +478,11 @@ impl<T: ArrowFloatType + Cosine + Dot + L2 + 'static> ProductQuantizer for Produ
 
                 // Normalized query vector.
                 let query = T::ArrayType::from(normalize(query.as_slice()).collect::<Vec<_>>());
-                self.l2_distance_table(&query, code)
+                // L2 over normalized vectors:  ||x - y|| = x^2 + y^2 - 2 * xy = 1 + 1 - 2 * xy = 2 * (1 - xy)
+                // Cosine distance: 1 - |xy| / (||x|| * ||y||) = 1 - xy / (x^2 * y^2) = 1 - xy / (1 * 1) = 1 - xy
+                // Therefore, Cosine = L2 / 2
+                let l2_dists = self.l2_distance_table(&query, code)?;
```

Comment:
Just confirming : ||x-y|| is L2 squared. So this means that the L2 distance in the table is actually the square of the distance right ?

Reply:
The L2_distance_table returned L2 squared

https://github.com/lancedb/lance/blob/27cf463365eaee35c94b087d6ad8884dc5e4068d/rust/lance-linalg/src/distance/l2.rs#L83